src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -187,7 +231,16 @@ public bool Lookup(DName name, out NameLookupInfo nameInfo, NameLookupPreference
                 if (_args.TryGetValue(name, out var value))
                 {
                     var type = value.TypeIdent.GetFormulaType()._type;
-                    nameInfo = new NameLookupInfo(BindKind.PowerFxResolvedObject, type, DPath.Root, 0, new UDFParameterInfo(type, value.ArgIndex, value.NameIdent.Name));
+
+                    if (_fvArgs != null)
+                    {
+                        var formulaValue = _fvArgs[value.ArgIndex];
+                        nameInfo = new NameLookupInfo(BindKind.PowerFxResolvedObject, type, DPath.Root, 0, formulaValue);
+                    }
+                    else
```

Comment:
Instead of changing here, make a change in EvalVisitor, Visit(ResolvedObjectNode node)

Reply:
Changes applied to EvalVisitor.